### PR TITLE
chore(techdocs-common): Improve deprecation warning and change temp folder creation

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

* Improve deprecation notice.  I saw this and couldn't (easily) identify which entity it was for:
  ```
  2021-03-04T17:42:58.181Z techdocs warn You are using the legacy git preparer in TechDocs which will be removed in near future (March 2021). Migrate to URL reader by updating `backstage.io/techdocs-ref` annotation in `catalog-info.yaml` to be prefixed with `url:`. Read the migration guide and benefits at https://github.com/backstage/backstage/issues/4409  type=plugin
  ```
* Now says:
  ```
  2021-03-04T17:42:58.181Z techdocs warn You are using the legacy git preparer in TechDocs for `backstage-poc` which will be removed in near future (March 2021). Migrate to URL reader by updating `backstage.io/techdocs-ref` annotation in `catalog-info.yaml` to be prefixed with `url:`. Read the migration guide and benefits at https://github.com/backstage/backstage/issues/4409  type=plugin
                                                                                                ^^^^^^^^^^^^^
                                                                                                ||||||||||||||
  ```
* I noticed my temp folders were being created as `/tmp/backstage-repo/mydomain.net/org/repo/branch/..` blah blah.  The `parseGitUrl()` method was identifying the domain of my git provider (mydomain.net) via `source` param, which is OK, but I think more sense to use the full host name `mygheinstance.mydomain.net` via `resource`. Esp if there could ever be two overlapping. (Rare?)

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4815-lpminbmztf498nmrtc13ncg6xe4vgfs6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60411cf1fcb22d18de6855a8

**Comment by adamdmharvey** [comment]
Action failure due to NPM download issue:

```
error An unexpected error occurred: "https://registry.yarnpkg.com/@types/serve-static/-/serve-static-1.13.9.tgz: Request failed \"500 Internal Server Error\"".
```

I posted something similar (cert issue) on Discord earlier; maybe I'm not going crazy! :-D

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4815-lpminbmztf498nmrtc13ncg6xe4vgfs6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60411cf1fcb22d18de6855a8

File: packages/techdocs-common/src/helpers.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export const getGitRepositoryTempFolder = async (
     // fs.realpathSync fixes a problem with macOS returning a path that is a symlink
     fs.realpathSync(os.tmpdir()),
     'backstage-repo',
-    parsedGitLocation.source,
+    parsedGitLocation.resource,
```
Comment (OrkoHunter) [inline packages/techdocs-common/src/helpers.ts:110]: Works 👌

**Review by OrkoHunter — APPROVED** [review]

File: packages/techdocs-common/src/helpers.ts (hunk 1 above)
Comment (freben) [reply packages/techdocs-common/src/helpers.ts:110]: Oh gosh this mistake has happened many times. Very confusing choice of field name on their part

**Pull request merged**